mutiny-core/src/storage.rs

```diff
@@ -0,0 +1,436 @@
+use crate::auth::AuthProfile;
+use crate::encrypt::{decrypt, encrypt};
+use crate::error::{MutinyError, MutinyStorageError};
+use crate::ldkstorage::CHANNEL_MANAGER_KEY;
+use crate::nodemanager::NodeStorage;
+use anyhow::anyhow;
+use bdk::chain::keychain::{KeychainChangeSet, KeychainTracker, PersistBackend};
+use bdk::chain::sparse_chain::ChainPosition;
+use bip39::Mnemonic;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+const KEYCHAIN_STORE_KEY: &str = "keychain_store";
+pub(crate) const MNEMONIC_KEY: &str = "mnemonic";
+const NODES_KEY: &str = "nodes";
+const AUTH_PROFILES_KEY: &str = "auth_profiles";
+const FEE_ESTIMATES_KEY: &str = "fee_estimates";
+
+fn needs_encryption(key: &str) -> bool {
+    match key {
+        MNEMONIC_KEY => true,
+        str if str.starts_with(CHANNEL_MANAGER_KEY) => true,
+        _ => false,
+    }
+}
```

Comment:
I think that'll be separate. This is encrypting just the values, when we do remote backups we'll want to encrypt the entire thing (keys and values) otherwise there will be a privacy leak.